src/Management/src/Endpoint/Actuators/HeapDump/HeapDumper.cs

```diff
@@ -45,7 +45,7 @@ public HeapDumper(IOptionsMonitor<HeapDumpEndpointOptions> optionsMonitor, strin
         {
             using var process = Process.GetCurrentProcess();
 
-            if (System.Environment.Version.Major == 3 || string.Equals("gcdump", _optionsMonitor.CurrentValue.HeapDumpType, StringComparison.OrdinalIgnoreCase))
+            if (string.Equals("gcdump", _optionsMonitor.CurrentValue.HeapDumpType, StringComparison.OrdinalIgnoreCase))
```

Comment:
I believe we wanted to change the default dump type to be `gcdump` for 4.0, though I'm not sure where the best place to set that is or what the current default is

Reply:
I removed this condition because it's redundant. [`Environment.Version`](https://learn.microsoft.com/en-us/dotnet/api/system.environment.version?view=net-8.0) returns the version of the CLR:
> Gets a version consisting of the major, minor, build, and revision numbers of the common language runtime.

Because Steeltoe targets .NET 8 and above, this is never true.